File: skellig-test-step-reader-sts/src/main/java/org/skellig/teststep/reader/sts/StsFileBufferedReader.java
package org.skellig.teststep.reader.sts;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.Reader;

class StsFileBufferedReader extends BufferedReader {

    StsFileBufferedReader(Reader in) {
        super(in);
    }

    void readUntilFindCharacter(char character) throws IOException {
        int c;
        while ((c = read()) > 0) {
            if (c == character) {
                break;
            }
        }
    }

}

File: skellig-test-step-reader-sts/src/main/kotlin/org/skellig/teststep/reader/sts/StsFileBufferedReader.kt
package org.skellig.teststep.reader.sts

import java.io.*

class StsFileBufferedReader(inputStream: InputStream) : BufferedInputStream(inputStream) {

    @Throws(IOException::class)
    fun readUntilFindCharacter(character: Char) {
        var c: Int
        while (read().also { c = it } > 0) {
            if (c == character.code) {
                break
            }
        }
    }
}